src/audio/generic_processor/generic_processor.c

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Marcin Rajwa <marcin.rajwa@linux.intel.com>
+
+/*
+ * A generic processor component.
+ */
+
+/**
+ * \file audio/generic_processor.c
+ * \brief Processing compoent aimed to work with external codecs
+ * \author Marcin Rajwa <marcin.rajwa@linux.intel.com>
+ */
+
+#include <sof/audio/buffer.h>
+#include <sof/audio/component.h>
+#include <sof/audio/pipeline.h>
+#include <sof/audio/generic_processor/generic_processor.h>
+#include <sof/common.h>
+#include <sof/debug/panic.h>
+#include <sof/drivers/ipc.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/clk.h>
+#include <sof/lib/memory.h>
+#include <sof/list.h>
+#include <sof/platform.h>
+#include <sof/string.h>
+#include <sof/ut.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <user/trace.h>
+
+static const struct comp_driver comp_generic_processor;
+
+//TODO: provide proper uuid below
+/* d8218443-5ff3-4a4c-b388-6cfe07b956aa */
+DECLARE_SOF_RT_UUID("generic_processor", gp_uuid, 0xd8218443, 0x5ff3, 0x4a4c,
+		    0xb3, 0x88, 0x6c, 0xfe, 0x07, 0xb9, 0x56, 0xaa);
+
+DECLARE_TR_CTX(gp_tr, SOF_UUID(gp_uuid), LOG_LEVEL_INFO);
+
+static struct comp_dev *generic_processor_new(const struct comp_driver *drv,
+					      struct sof_ipc_comp *comp)
+{
+	int ret;
+	struct comp_dev *dev = NULL;
+	struct comp_data *cd = NULL;
+
```

Comment:
No need to initialize these two variables with NULL.

Reply:
These are auto variables, if we fail at the beginning of.new() and go to error handling code section the `rfree` will try to reclaim this memory. So the NULL assignment makes sure we don't call `rfree` if this particular object has not been allocated.

Reply:
I think for dev it's still unnecessary since there is no branch between these initializations and the actual allocation. But yeah, better safe I guess.

Reply:
@plbossart, and this is exactly what this code does.

Reply:
no, you need to have different labels and no conditional checks.

Reply:
this will make code less readable I think, why are you afraid of conditional checks in error handling procedure? This is not sth which is called every milisecods so we shouldn't be concerned about performance at all...if that's what you care about

Reply:
that's how driver code is written, it helps make sure the resources are released in the right order. 
Testing for NULL variables before freeing them is not what we do.